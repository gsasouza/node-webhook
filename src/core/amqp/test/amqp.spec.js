const testSetup = require('../../../test/setup');
const { consume, publish } = require('../amqp');

const {
  getQueues,
  clearQueues,
} = require('../../../test/mock/amqplib');

beforeAll(() => testSetup());

beforeEach(() => clearQueues());

test('should publish a message to queue', async () => {
  await publish(1, {
    webhook: 'https://localhost.com:5003',
    payload: {
      message: 'test',
    }
  });
  expect(getQueues()).toMatchSnapshot();
});

test('should consume a message from a queue', async () => {
  await publish(1,
    {
      payload: {
        message: 'test',
      },
      webhook: 'https://localhost.com:5003',
      tryTimes: 0,
      updatedAt: Date.now(),
    }
  );
  const message = await consume(1);
  expect(message).toMatchSnapshot();
  expect(getQueues()).toMatchSnapshot();
});

test('should return undefined if does not have messages to consume', async () => {
  const message = await consume(1);
  expect(message).toMatchSnapshot();
});
