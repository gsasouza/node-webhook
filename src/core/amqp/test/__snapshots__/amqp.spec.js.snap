// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should consume a message from a queue 1`] = `
Object {
  "payload": Object {
    "message": "test",
  },
  "tryTimes": 0,
  "updatedAt": 1,
  "webhook": "https://localhost.com:5003",
}
`;

exports[`should consume a message from a queue 2`] = `
Object {
  "messageCount": 0,
  "queue": "1",
}
`;

exports[`should publish a message to queue 1`] = `
Object {
  "messageCount": 1,
  "queue": "1",
}
`;
